jbpm-flow/src/main/java/org/jbpm/process/core/timer/impl/QuartzSchedulerService.java

```diff
@@ -156,8 +159,16 @@ public void internalSchedule(TimerJobInstance timerJobInstance) {
         jobq.setRequestsRecovery(true);
         jobq.getJobDataMap().put("timerJobInstance", timerJobInstance);
             
+        // Amend nextFireTime not to schedule older than now + RESCHEDULE_DELAY
+        Date nextFireTime = timerJobInstance.getTrigger().hasNextFireTime();
+        Date threshold = new Date(System.currentTimeMillis() + RESCHEDULE_DELAY);
+        if (nextFireTime.before(threshold)) {
```

Comment:
In the previous commit, it was nextFireTime.before(now). But now I make sure that the nextFireTime will be later than (now+RESCHEDULE_DELAY).